### PR TITLE
updating Firebase on App Engine sample to use latest libs

### PR DESCRIPTION
**Pull request by jabubake**
Fixing issue in running app locally as reported by customer:
https://cloud.google.com/appengine/docs/deprecations/channel

### CODE REVIEW

File: appengine/firebase-tictactoe/README.md [hunk 1]
```diff
@@ -29,26 +23,36 @@ for realtime notifications when the board changes.
 [fb-overview]: https://console.firebase.google.com/project/_/overview
 [maven]: https://maven.apache.org
 
+## Setup
+
+* If you haven't already, Download and initialize the [Cloud SDK](https://cloud.google.com/sdk/)
```
Comment (lesv) [inline appengine/firebase-tictactoe/README.md]: Shouldn't this be a Prerequisite?

File: appengine/firebase-tictactoe/README.md [hunk 2]
```diff
@@ -29,26 +23,36 @@ for realtime notifications when the board changes.
...
+```
+    gcloud init
+```
+* If you haven't already, Create an App Engine app within the current Google Cloud Project
```
Comment (lesv) [inline appengine/firebase-tictactoe/README.md]: Redundant phrase

File: appengine/firebase-tictactoe/README.md [hunk 3]
```diff
@@ -29,26 +23,36 @@ for realtime notifications when the board changes.
...
+```
+    gcloud app create
+```
+* If you haven't already, setup
+[Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials)
+to run your application locally:
+Download [service account credentials][creds] and set the `GOOGLE_APPLICATION_CREDENTIALS`
+environment variable to its path:
+```
+   export GOOGLE_APPLICATION_CREDENTIALS=/path/to/your/credentials.json
```
Comment (lesv) [inline appengine/firebase-tictactoe/README.md]: Should we have them set GOOGLE_CLOUD_PROJECT?  Look at gcloud config to verify their project?

File: appengine/firebase-tictactoe/src/test/java/com/example/appengine/firetactoe/MoveServletTest.java [hunk 4]
```diff
@@ -136,12 +136,13 @@ public LowLevelHttpResponse execute() throws IOException {
...
     });
-    FirebaseChannel.getInstance().httpTransport = mockHttpTransport;
+    FirebaseChannel.getInstance(null).httpTransport = mockHttpTransport;
 
+    Mockito.doReturn(null).when(servletUnderTest).getServletContext();
     servletUnderTest.doPost(mockRequest, mockResponse);
 
     game = ofy.load().type(Game.class).id(gameKey).safe();
-    assertThat(game.board).isEqualTo(" X       ");
+    assertEquals(game.board, " X       ");
```
Comment (lesv) [inline appengine/firebase-tictactoe/src/test/java/com/example/appengine/firetactoe/MoveServletTest.java]: Truth might make this cleaner.

File: appengine/firebase-tictactoe/README.md (hunk 2 above)
Comment (jabubake) [reply appengine/firebase-tictactoe/README.md]: cleaned up

File: appengine/firebase-tictactoe/README.md (hunk 1 above)
Comment (jabubake) [reply appengine/firebase-tictactoe/README.md]: done

File: appengine/firebase-tictactoe/README.md (hunk 3 above)
Comment (jabubake) [reply appengine/firebase-tictactoe/README.md]: given we still ask users to use `gcloud init` : added a note there that this sets the project.

File: appengine/firebase-tictactoe/src/test/java/com/example/appengine/firetactoe/MoveServletTest.java (hunk 4 above)
Comment (jabubake) [reply appengine/firebase-tictactoe/src/test/java/com/example/appengine/firetactoe/MoveServletTest.java]: Minimizing test dependencies, so leaving this as-is

**Pull request merged**